main/com/collections/Methods.java

```diff
@@ -137,4 +137,91 @@ public boolean canBePalindrome(String s) {
 
         return arraysSum;
     }
+
+    private void reverseArray(int[] array) {
+        for (int i = 0; i < array.length / 2; i++) {
+            int temp = array[i];
+            array[i] = array[array.length - i - 1];
+            array[array.length - i - 1] = temp;
+        }
+    }
+
+    public int[] sumTwoIntArrays1(int[] array1, int[] array2) {
+
+        // reverse Arrays with private method
+        reverseArray(array1);
+        reverseArray(array2);
+
+        // crate array to store sum
+        ArrayList<Integer> sumOfArrays = new ArrayList<>();
+
+
+        int nextDigit = 0;
+
+        //find which array is bigger and which smaller
+        int lessDigitsAmount = Math.min(array1.length, array2.length);
+        int moreDigitsAmount = Math.max(array1.length, array2.length);
+
+        //first adding part (the same index value in arrays)
+        for (int i = 0; i < lessDigitsAmount; i++) {
+            if ((array1[i] + array2[i] + nextDigit) > 9) {
```

Comment:
You can assign **array1[i] + array2[i] + nextDigit** to some local value and reuse in 3 places.